worktrolly-ui/src/app/services/auth.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import firebase from 'firebase/app';
+import { AngularFireAuth } from '@angular/fire/auth';
+import { AngularFirestore, AngularFirestoreDocument } from '@angular/fire/firestore';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+
+  loginForm: FormGroup = new FormGroup({
+    email: new FormControl("", [Validators.required, Validators.email]),
+    password: new FormControl("", [Validators.required, Validators.minLength(6)]),
+  });
+  signupForm: FormGroup = new FormGroup({
+    username: new FormControl("", [Validators.required, Validators.minLength(3)]),
+    email: new FormControl("", [Validators.required, Validators.email]),
+    password: new FormControl("", [Validators.required, Validators.minLength(6)]),
+  });
+  resetLoginForm() {
+    this.loginForm.setValue({
+      email: "",
+      password: ""
+    })
+  }
+  resetSignupForm() {
+    this.signupForm.setValue({
+      username: "",
+      email: "",
+      password: ""
+    })
+  }
+
+  constructor(public afauth: AngularFireAuth, public afs: AngularFirestore) { }
+
+  private createUserData(user) {
+    // Sets user data to firestore on login
+    const userRef: AngularFirestoreDocument<any> = this.afs.doc(
+      `Users/${user.uid}`
+    );
+
+    const data = {
+      uid: user.uid,
+      email: user.email,
+      displayName: user.displayName,
```

Comment:
It's good  to use the interface for this data

Reply:
Even if the interface is created, whenever login is triggered, we need an observable to read the user data, and store it into the application.